spec/javascripts/actions/notifications_spec.js

```diff
@@ -0,0 +1,199 @@
+import status from 'http-status';
+
+describe('Notifications Actions', () => {
+  describe('sendSuccessNotification', () => {
+    it('dispatches addNotification with the new notification', () => {
+      const newNotification = {
+        id: 42,
+        type: types.SUCCESS,
+        message: 'Success message'
+      }
+
+      const FakeNotification = {
+        createNotification: () => newNotification,
+        types
+      };
+
+      const fakeDispatch = sinon.stub();
+
+      Notification.sendSuccessNotification(newNotification.message)
+        (fakeDispatch, null, { Notification: FakeNotification });
+
+      expect(fakeDispatch).toHaveBeenCalledWith(Notification.addNotification(newNotification));
+    });
+
+    it('dispatches removeNotification after 5000ms to remove notification', () => {
+      const newNotification = {
+        id: 42,
+        type: types.SUCCESS,
+        message: 'Success message'
+      }
+
+      const FakeNotification = {
+        createNotification: () => newNotification,
+        types
+      };
+
+      const fakeDispatch = sinon.stub();
+
+      Notification.sendSuccessNotification(newNotification.message)
+        (fakeDispatch, null, { Notification: FakeNotification });
+
+      setTimeout(() => {
+        expect(fakeDispatch).toHaveBeenCalledWith(Notification.removeNotification(newNotification.id));
+      }, 5000);
+    });
+  });
+
+  describe('sendErrorNotification', () => {
+    const createNotification = (type, message) => ({
+      id: 42,
+      type,
+      message
+    });
+
+    describe('when respose error is an unprocessable_entity', () => {
+      const error = {
+        response: {
+          status: status.UNPROCESSABLE_ENTITY,
+          data: {
+            story: {
+              errors: {}
+            }
+          }
+        }
+      };
+
+      it('dispatches addValidationNotifications', () => {
+        const FakeNotification = {
+          createNotification,
+          types
+        };
+
+        const fakeDispatch = sinon.stub();
+
+        Notification.sendErrorNotification(error)
+          (fakeDispatch, null, { Notification: FakeNotification });
+
+        expect(fakeDispatch).toHaveBeenCalled();
+      });
+    });
+
+    describe('when respose error is unauthorized', () => {
+      const error = {
+        response: {
+          status: status.UNAUTHORIZED
+        }
+      };
+
+      it('dispatches addNotification with the unauthorized message', () => {
+        const FakeNotification = {
+          createNotification,
+          types
+        };
+
+        const newNotification = createNotification({
+          type: types.ERROR,
+          message: I18n.t('users.You are not authorized to perform this action')
+        });
+
+        const fakeDispatch = sinon.stub();
+
+        Notification.sendErrorNotification(error)
+          (fakeDispatch, null, { Notification: FakeNotification });
```

Comment:
Avoid breaking this line like this, it makes it confusing (for both the JS interpreter and for the reader). Prefer doing like his:

```suggestion
        Notification.sendErrorNotification(error)(
          fakeDispatch, null,
          { Notification: FakeNotification }
        );
```

Reply:
It applies to the other occurrences as well.